### ISSUE TITLE
Subpixel smoothing for conductivities

### ISSUE THREAD

**Issue by smartalecH** [NanoComp/meep#2334]
We have subpixel smoothing for purely dielectric `MaterialGrids`, and we have [adjoint support for conductivities](https://github.com/NanoComp/meep/blob/9473d3117e8a725488728cdeaaecb0e8c9ac3671/src/meepgeom.cpp#L2911). What would it take to enable subpixel smoothing for conductivities (specifically within a `MaterialGrid`)?

In particular, what would it take to allow for an infinite scaling of $\beta$ when we have conductivities (if smoothing conductivities is out of the question, can we still enable this)?

In other words, it would be great to be able to smoothly optimize lossy materials/metals too.

(Let's assume no polarizabilities, e.g. dispersive profiles, since we don't do any smoothing for those).

**Comment by stevengj** [comment]
To get a second-order accurate method is hard because we don't currently support arbitrary anisotropic conductivity.

However, if we just do *isotropic* smoothing (1st-order) accurate, it should be relatively easy.   It shouldn't be any worse in accuracy than doing no smoothing at all, and has the advantage of supporting β=∞.
